src/core/lib/security/transport/security_handshaker.cc

```diff
@@ -652,14 +655,17 @@ class ServerSecurityHandshakerFactory : public HandshakerFactory {
 //
 
 RefCountedPtr<Handshaker> SecurityHandshakerCreate(
-    tsi_handshaker* handshaker, grpc_security_connector* connector,
-    const ChannelArgs& args) {
+    absl::StatusOr<tsi_handshaker*> handshaker,
+    grpc_security_connector* connector, const ChannelArgs& args) {
   // If no TSI handshaker was created, return a handshaker that always fails.
   // Otherwise, return a real security handshaker.
-  if (handshaker == nullptr) {
-    return MakeRefCounted<FailHandshaker>();
+  if (!handshaker.ok()) {
+    return MakeRefCounted<FailHandshaker>(handshaker.status());
```

Comment:
Please change `handshaker.status()` to:

```
absl::Status(
    handshaker.status().code(),
    absl::StrCat("Failed to create security handshaker: ", handshaker.status().message())
```

This way, we're adding a useful prefix to the status message but leaving the code as-is.